boards/adafruit-itsy-bitsy-rp2040/examples/adafruit_itsy_bitsy_rainbow.rs

```diff
@@ -66,12 +64,12 @@ fn main() -> ! {
     );
 
     let mut n: u8 = 128;
+    let mut timer = timer;
```

Comment:
The `mut` can be placed on the first binding :)

Reply:
If I change `let timer = Timer::new(pac.TIMER, &mut pac.RESETS, &clocks);` to `let mut timer = Timer::new(pac.TIMER, &mut pac.RESETS, &clocks);`

Then compiler complains with the following error:

```text
error[E0502]: cannot borrow `timer` as mutable because it is also borrowed as immutable
  --> boards/adafruit-itsy-bitsy-rp2040/examples/adafruit_itsy_bitsy_rainbow.rs:72:9
   |
63 |         timer.count_down(),
   |         ----- immutable borrow occurs here
...
69 |         ws.write(brightness(once(wheel(n)), 32)).unwrap();
   |         -- immutable borrow later used here
...
72 |         timer.delay_ms(25);
   |         ^^^^^^^^^^^^^^^^^^ mutable borrow occurs here

```

Reply:
oh I see. It might be worth adding a note that the "rebind" is to force a copy of the timer.
Or maybe change `63 | timer.count_down(),` to `63 | timer.clone().count_down(),` ?

Reply:
`timer.clone().count_down()` fails with an error `creates a temporary value which is freed while still in use`, so I added a note